packages/@aws-cdk/aws-lambda/lib/code.ts

```diff
@@ -51,6 +51,24 @@ export abstract class Code {
     return this.fromAsset(path);
   }
 
+  /**
+   * Provides a code asset by building a Dockerfile and running it. The docker
+   * image should copy its code to /asset-output when it is run.
```

Comment:
This is really interesting. I'm wondering if there's something to do in `@aws-cdk/core` here instead.

If the asset is built by building a Dockerfile then why should we mount `/asset-input` or run the container? Volume mounts are heavy in terms of performance (see also issues with macOS).

If the asset is created at a well known location then we can do something like this after building the Dockerfile:

```bash
id=$(docker create image-name-from-build) # does not start the container
docker cp $id:/asset-output cdk.out/xxx # copy asset to staging dir
docker rm -v $id # remove container
```

Reply:
That's a very good point! If we were to add a new member to `BundlingOptions`, like `imageStaticAssetPath` (or something like that), we could handle the docker create, cp, and rm from the `bundle` function in `AssetStaging`.

Reply:
> That's a very good point! If we were to add a new member to `BundlingOptions`, like `imageStaticAssetPath` (or something like that), we could handle the docker create, cp, and rm from the `bundle` function in `AssetStaging`.
> 
> A reason for `imageStaticAssetPath` over any specific path would be the case where a user wants one docker build that emits several different assets - each bundler pass may use the same image, but would access different directories in the image. But, it makes a lot of sense to set a default path like `/asset-output`.

A new commit implementing this idea is now up.